internal/component/remote/vault/vault_test.go

```diff
@@ -190,7 +192,10 @@ func getTestVaultServer(t *testing.T) *vaultapi.Client {
 	require.NoError(t, err)
 
 	t.Cleanup(func() {
-		require.NoError(t, container.Terminate(ctx))
+		terminateErr := container.Terminate(ctx)
+		if !strings.Contains(terminateErr.Error(), "context canceled") { // avoid race condition on test context cancelled
```

Comment:
Why is this necessary? Is it because the container terminates async? Do we need to wait after the Terminate function? Even if it's async I don't see how it can cause an issue, since the context should only cancel at the end of the test, after Terminate has already completed.

Reply:
functions registered with `t.Cleanup` are called in registration order. Context is created from [TestContext](https://github.com/grafana/alloy/blob/main/internal/runtime/componenttest/context.go#L9). This context will be canceled when the test if finished and it happens before this cleanup hook is called. We chould just pass a next context here instead of reusing the existing one

Reply:
Because it is failing here: https://drone.grafana.net/grafana/alloy/7212/1/2 
You may need to ask the author of the test, right now I would like to fix tests in main.

Reply:
Never mind not sure this is true